src/databricks/labs/ucx/contexts/application.py

```diff
@@ -391,7 +395,7 @@ def file_resolver(self):
 
     @cached_property
     def dependency_resolver(self):
-        import_resolvers = [self.file_resolver, self.whitelist_resolver]
+        import_resolvers = [self.pip_resolver, self.file_resolver, self.whitelist_resolver]
```

Comment:
The pip resolver is NOT an import resolver. You need to add a _library_resolver member in DependencyResolver with type BaseLibraryResolver

Reply:
And PipResolver should derive from BaseLibraryResolver